tools/flang2/flang2exe/ll_structure.c

```diff
@@ -578,6 +579,21 @@ ll_create_module(const char *module_name, const char *target_triple,
 
   compute_ir_feature_vector(new_module, llvm_ir_version);
   compute_datalayout(new_module);
+   
+  #ifdef _WIN32
+
+  if (flg.linker_directives) {
+    int i;
+    char *linker_directive;
+    LLMD_Builder mdb = llmd_init(new_module);
+    for (i = 0; (linker_directive = flg.linker_directives[i]); ++i) {
+      llmd_add_string(mdb, linker_directive);
+    }
+    LL_MDRef linker_md = llmd_finish(mdb);
+    ll_extend_named_md_node(new_module, MD_llvm_linker_options, linker_md);
+  }
```

Comment:
For 4.0, something like this:

```
LLMD_Builder mdb = llmd_init(new_module);
for (i = 0; (linker_directive = flg.linker_directives[i]); ++i) {
    LLMD_Builder submdb = llmd_init(new_module);
    llmd_add_string(submdb, linker_directive);
    LL_MDRef submd = llmd_finish(submdb);
    llmd_add_md(mdb, submd);
}
LL_MDRef md = llmd_finish(mdb);

LLMD_Builder boilerplatemdb = llmd_init(new_module);
llmd_add_i32(boilerplatemdb, 6);
llmd_add_string(boilerplatemdb, "Linker Options");
llmd_add_md(boilerplatemdb, md);
LL_MDRef boilerplatemd = llmd_finish(boilerplatemdb);

ll_extend_named_md_node(new_module, MD_llvm_module_flags, linker_md);
```

What a load of nonsense!

Reply:
I'm getting `warning: ignoring debug info with an invalid version (0) in C:\Users\idf2\AppData\Local\Temp\hello_world-445182.ll`

Reply:
What's the LLVM generated?

Reply:
`ps: while ($true) { cp "$env:TMP\*.ll" C:\Downloads }`

Reply:
`lldbg_init` will initialize the debug info (inc. the version). but are there other side effects?